app/helpers/annotation_categories_helper.rb

```diff
@@ -3,16 +3,46 @@ module AnnotationCategoriesHelper
   def prepare_for_conversion(annotation_categories)
     result = {}
     annotation_categories.each do |annotation_category|
-      annotation_texts = []
-      annotation_category.annotation_texts.each do |annotation_text|
-        annotation_texts.push(annotation_text.content)
+      if annotation_category.flexible_criterion.nil?
+        annotation_texts = []
+        annotation_texts.push(nil)
+        annotation_category.annotation_texts.each do |annotation_text|
```

Comment:
I know you didn't write this part, but take the opportunity to simplify using `pluck` to immediately return an array.